flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/MailboxExecutor.java

```diff
@@ -227,21 +226,4 @@ default void execute(@Nonnull Runnable command, String description) {
 	 * @throws IllegalStateException if the mailbox is closed and can no longer supply runnables for yielding.
 	 */
 	boolean tryYield();
-
-	/**
-	 * Provides an {@link Executor} view on this {@code MailboxExecutor}, where submitted tasks will receive the
-	 * given description. The {@link Executor} can be used with {@link java.util.concurrent.CompletableFuture}.
-	 *
-	 * <p>An optional description can (and should) be added to ease debugging and error-reporting. The description
-	 * may contain placeholder that refer to the provided description arguments using {@link java.util.Formatter}
-	 * syntax. The actual description is only formatted on demand.
-	 *
-	 * @param descriptionFormat the optional description for all commands that is used for debugging and
-	 * error-reporting.
-	 * @param descriptionArgs the parameters used to format the final description string.
-	 * @return an {@code Executor} view on this {@code MailboxExecutor}
-	 */
-	default Executor asExecutor(String descriptionFormat, Object... descriptionArgs) {
```

Comment:
FYI, having a bridge to java...Executor was meant to help implementing CompletableFutures. However, since the interface is so simple, it might as well be a lambda implementation when it's actually used in CompletableFutures.

Reply:
Looking at the last commit, will actually make the mentioned adhoc implementation quite verbose. If you think CompleteableFutures over mailbox is a valid use case, reconsider removing this method.

Reply:
It's not used currently anywhere, so I prefered to drop it instead of implementing `Runnable` -> `RunnableWithException` wrapping of the `Executor` interface. Now I see that that's only method (previously I thought it's more), but still - I would prefer to not maintain a dead code. We can re-introduce it later if there will be such demand.